pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractIgnoredAnnotRule.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.rule;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import net.sourceforge.pmd.properties.StringMultiProperty;
+
+public class AbstractIgnoredAnnotRule extends AbstractJavaRule {
+
+    protected final StringMultiProperty ignoredAnnotationsDescriptor
+        = StringMultiProperty.named("ignoredAnnotations")
+        .desc("Fully qualified names of the annotation types that should be ignored by this rule")
+        .defaultValues(defaultSuppressionAnnotations())
+        .build();
+
```

Comment:
The property should probably be declared in a protected constructor here

Reply:
I didn't catch your meaning. Is there any advantage in doing this?

Reply:
Yes, if you do that, subclasses will not need to write a constructor like so:
```java
 XXRule() {
  definePropertyDescriptor(ignoredAnnotationsDescriptor);
}
```

The idea is to add this line to a noargs protected constructor in the abstract class, so that it's automatically called before subclass construction.

Reply:
I get it.